src/sentry/ingest/userreport.py

```diff
@@ -93,10 +107,119 @@ def find_event_user(report_data, event):
             return None
 
     tag = event.get_tag("sentry:user")
+
     if not tag:
         return None
 
     try:
         return EventUser.for_tags(project_id=report_data["project_id"], values=[tag])[tag]
     except KeyError:
         pass
+
+
+def find_event_user_with_snuba(event: Event, eventuser: EventUser):
+    """
+    Query Snuba to get the EventUser information for an Event.
+    Then compare with the EventUser record and confirm that the
+    query result is equivalent.
+    """
+    start_date, end_date = _start_and_end_dates(event.datetime)
+
+    query = _generate_entity_dataset_query(
+        event.project_id, event.group_id, event.event_id, start_date, end_date
+    )
+    request = Request(
+        dataset=Dataset.Events.value,
+        app_id=REFERRER,
+        query=query,
+        tenant_ids={"referrer": REFERRER, "organization_id": event.project.organization.id},
+    )
+    data_results = raw_snql_query(request, referrer=REFERRER)["data"]
+
+    if len(data_results) == 0:
+        return logger.info(
+            "Errors dataset query to find EventUser did not return any results.",
+            extra={
+                "eventuser": serialize(eventuser),
+                "event_id": event.event_id,
+                "project_id": event.project_id,
+                "group_id": event.group_id,
+            },
+        )
+
+    if not _is_equal_to_eventuser(eventuser, data_results[0]):
+        return logger.info(
+            "EventUser not the same as Errors dataset query.",
+            extra={
+                "eventuser": serialize(eventuser),
+                "dataset_results": data_results[0],
+                "event_id": event.event_id,
+                "project_id": event.project_id,
+                "group_id": event.group_id,
+            },
+        )
+
+    else:
+        return logger.info(
+            "EventUser equal to results from Errors dataset query.",
+            extra={
+                "event_id": event.event_id,
+                "project_id": event.project_id,
+                "group_id": event.group_id,
+            },
+        )
+
+
+def _is_equal_to_eventuser(eventuser: EventUser, snuba_result: Mapping[str, Any]):
+    EVENTUSER_TO_SNUBA_KEY_MAP = {
+        "project_id": ["project_id"],
+        "ident": ["user_id"],
```

Comment:
`ident` can be any of these fields https://github.com/getsentry/sentry/blob/f25519226d4345343c0539f9e3ac49de57c576d0/src/sentry/models/eventuser.py#L16-L23.

Not sure if `user_id` already encapsulates this though...